flask/json/__init__.py

```diff
@@ -20,6 +20,11 @@
 # depend anyways.
 from itsdangerous import json as _json
 
+try:
+    import dataclasses
+    dataclass_available = True
+except ImportError:
+    dataclass_available = False
```

Comment:
What about just setting `dataclasses = None` in this case and then also using that as a criterion on whether they are available or not?

Reply:
Done